pkg/cli/admin/upgrade/status/status.go

```diff
@@ -246,8 +258,6 @@ func (o *options) Run(ctx context.Context) error {
 		}
 	}
```

Comment:
I 'm not a fan of doing all iteration twice, feels like we can improve the algorithm here so it is even a bit more readable (the new modal behavior of `selectNodesFromPool` also seems unintuitive.

I think we can do something like the following:

1. Iterate over pools and find worker pool, master pool, and custom pools. Actually, we can get away of with just finding their selectors?
2. Make a function that takes a pool and a node and returns whether the node matches
3. Make a function that takes all the pool (separated by (1)) and a node returns the name of the pool the node. We can use the simplified hierarchy because we do not care about errors. If a node is covered by master, it is a master. Else if it is covered by at least one custom, it's that custom. Else if its covered by worker, its a worker. Otherwise it belongs to no pool.
4. Iterate over all nodes, get the matching pool for every one of them, build an index of nodes per each pool


```
var masterSelector ...the type that v1.LabelSelectorAsSelector returns...
var workerSelector ..
var customSelectors map[string]...the type that v1.LabelSelectorAsSelector returns...

selector, err := 
for pool in pools:
  switch pool.name {
    // we'd need to handle errors ofc
    master: master = v1.LabelSelectorAsSelector(pool.Spec.NodeSelector)
    worker: worker = v1.LabelSelectorAsSelector(pool.Spec.NodeSelector)
    default: customSelectors[pool.name] = v1.LabelSelectorAsSelector(pool.Spec.NodeSelector)
  }

func whichPool(master, worker MCP // actually we need just selectors, custom []MCP, node Node // actually we need just the labels) string {
  if master.Matches(labels.Set(node.Labels)) return "master"
  for customSelector in custom:
    if customSelector.Matches(labels.Set(node.Labels)): return pool.name
  if worker.Matches(labels.Set(node.Labels)): return "worker"
  return ""
}

var nodesPerPool map[name][]Node

for node in nodes {
  name = whichPool(master, worker, custom, node)
  nodesPerPool[name] = append(nodesPerPool[name], node)
}

// and now you can start assessing individual pools

for _, pool := range pools.Items {
    nodesStatusData, insights := assessNodesStatus(cv, pool, nodesPerPool[pool.Name], machineConfigs.Items)
   ...
```